src/test/java/com/coherent/task40/DownloadTest.java

```diff
@@ -0,0 +1,91 @@
+package com.coherent.task40;
+
+import org.junit.Ignore;
+import org.junit.jupiter.api.*;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Dimension;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.chrome.ChromeDriver;
+import org.openqa.selenium.devtools.v106.overlay.model.LineStyle;
+import org.openqa.selenium.interactions.Actions;
+import org.openqa.selenium.support.ui.ExpectedConditions;
+import org.openqa.selenium.support.ui.Select;
+import org.openqa.selenium.support.ui.WebDriverWait;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Pattern;
+
+public class DownloadTest {
+    WebDriver driver;
+
+
+    @BeforeEach
+    void setup() {
+        driver = new ChromeDriver();
+        driver.manage().timeouts().implicitlyWait(5, TimeUnit.SECONDS);
+        driver.manage().window().setSize(new Dimension(1700, 1000));
+    }
+
+    @Test
+    public void download50Percentage() {
+        By downloadButton = By.id("cricle-btn");
+        By percentText = By.cssSelector(".percenttext");
+        String expectedPercent = "50%";
+        String zeroPercent = "0%";
+        String source = "https://demo.seleniumeasy.com/bootstrap-download-progress-demo.html";
```

Comment:
static info like URLs should be moved to one place. It could be property file or to class with constants. Class name could TestConstants

Reply:
Done. Created separate class BaseTest and added constants there. Plus added URL to the BaseTest class in the setup() method